src/Service/PaymentRestrictionValidation/ApplePayPaymentRestrictionValidation.php

```diff
@@ -71,4 +67,15 @@ private function isIosDevice()
 
         return (bool) $this->context->getMobileDetect()->is('ios');
     }
+
+    private function isMacDesktop()
+    {
+        if (SaferPayConfig::isTestMode()) {
+            return true;
+        }
+
+        $device = $this->context->getDeviceDetect();
+
+        return $device === 1 && strpos($_SERVER['HTTP_USER_AGENT'], 'Macintosh') !== false;
```

Comment:
Unsure what "1" is, Could use \Context::DEVICE_COMPUTER.

preg_match('/macintosh|mac os x|mac_powerpc/i', $_SERVER['HTTP_USER_AGENT']). Should  be the way to check if it's Mac desktop